packages/compass-connections/src/stores/connections-store.ts

```diff
@@ -413,21 +395,16 @@ export function useConnections({
               // Get updated secrets first (and not in parallel) so that the
               // race condition window between load() and save() is as short as possible.
               const mergeConnectionInfo = {
+                id: connectionInfo.id,
                 connectionOptions: await dataService.getUpdatedSecrets(),
-              };
+              } as ConnectionInfo;
```

Comment:
This is kind of a valid complaint from TS, right? Should `saveConnectionInfo` accept a `Partial<ConnectionInfo>` (or `Partial<ConnectionInfo> & Pick<ConnectionInfo, 'id'>`)?